src/cms/static/js/pages/page_bulk_actions.js

```diff
@@ -2,44 +2,58 @@
-        selected_pages.forEach(e => e.checked = true);
-    } else {
-        selected_pages.forEach(e => e.checked = false);
+u(document).on('DOMContentLoaded', function () {
+    
+    u('[name="select_all_pages"]').on('click', select_all_pages);
+    function select_all_pages() {
+        var selected_pages = document.getElementsByName("page_selected");
+        if (document.getElementsByName("select_all_pages")[0].checked == true) {
+            selected_pages.forEach(e => e.checked = true);
+        } else {
+            selected_pages.forEach(e => e.checked = false);
+        }
     }
-}
 
-function bulk_action_execute() {
-    var action = document.getElementById('bulk_action').value;
-    // match action types
-    if ( action == "archive_pages" ) {
-        // TODO
-    } else if (action == "pdf_export") {
-        var selected_pages = document.getElementsByName("page_selected");
-        pages = [];
-        selected_pages.forEach(e => {
-            if(e.checked) {
-                pages.push(e.value)
-            }
-        })
-        if (pages.length != 0) {
-            var pdf_export_url = document.getElementById("pdf_export_url").value;
-            window.open(pdf_export_url+"?pages="+pages.join(','));
-        } 
-    } else { // no previous match, than language code -> XLIFF export
-        var page_selected = document.getElementsByName("page_selected");
+    u("#bulk_action_btn").on('click', bulk_action_execute);
+    function bulk_action_execute() {
+        var action = u('#bulk_action').first().value;
         var pages = [];
-        page_selected.forEach(async function (page_checkbox) {
-            if(page_checkbox.checked) {
-                pages = pages.concat(page_checkbox.value);
-            }
-        });
+        u('input').filter(':checked').each((node, i) => pages.push(node.value));
         if (pages.length == 0) {
+            // abort if no pages selected
             return;
         }
-        var bulk_action_url = document.getElementById("bulk_action_url").value;
-        window.open(bulk_action_url+"?target_lang="+action+"&pages="+pages.join(','), "_blank");
+        // match action types
+        if ( action == "archive_pages" ) {
+            // TODO
+        } else if (action == "pdf_export") {
+            var pdf_export_url = document.getElementById("pdf_export_url").value;
+            window.open(pdf_export_url+"?pages="+pages.join(','));
+        } else { // no previous match, than language code -> XLIFF export
+            const state = true;
+            const url = u("#ajax_url").first().dataset.url;
+            var bulk_action_url = document.getElementById("bulk_action_url").value;
+            // download XLIFF
+            window.open(bulk_action_url+"?target_lang="+action+"&pages="+pages.join(','), "_blank");
+            // update state of translations to currently_in_translation
+            post_translation_state(url, state, pages, action)
```

Comment:
This seems inefficient to me. We could set the translation state during the export instead of sending a second HTTP request to the server. However, I see the advantage of having just one mechanism for setting the state. This could be done somewhere around https://github.com/Integreat/integreat-cms/blob/develop/src/cms/views/pages/page_actions.py#L313

Reply:
Yes, the reason for two requests was, that one responses a zip file while the other responses those selected pages, that actually have an existing translation for updating the icon to "translation in progress". But of course I can circumvent that second one by saving all existing translation languages of a page somewhere in the DOM and just take that value. That would halve the number of requests and even remove one of the scattered js files, which I added for this PR.